garrysmod/addons/[brick]bricks_essentials/lua/entities/bricks_server_rock/init.lua

```
@@ -2,81 +2,299 @@ AddCSLuaFile("cl_init.lua")
+	else
+		ErrorNoHalt("[Brick's Server] ERROR: Failed to initialize physics for rock entity at " .. tostring(self:GetPos()))
 	end
+end
 
-	self:GetPhysicsObject():EnableMotion( false )
-
-	local rockTable = BRICKS_SERVER.CONFIG.CRAFTING.RockTypes[self.rockType or ""]
-	if( rockTable ) then
-		self:SetRockType( self.rockType or "" )
+-- Initialize rock type data with validation
+function ENT:InitializeRockData()
```

Luacheck (1.2.0)

[warning] mutating non-standard global variable 'ENT'

(W112)